apps/io-wallet-user-func/src/email-provider.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  MailerTransporter,
+  sendMail,
+} from "@pagopa/io-functions-commons/dist/src/mailer";
+import * as RTE from "fp-ts/lib/ReaderTaskEither";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as t from "io-ts";
+
+import { MailConfig } from "./app/config";
+
+export const SendEmailParams = t.type({
+  html: t.string,
+  subject: t.string,
+  text: t.string,
+  to: t.string,
+});
+
+export type SendEmailParams = t.TypeOf<typeof SendEmailParams>;
+
+export const sendEmail: (
+  transporter: MailerTransporter,
+  params: SendEmailParams,
+) => RTE.ReaderTaskEither<{ mail: MailConfig }, Error, void> =
+  (transporter, { html, subject, text, to }) =>
+  (configs) =>
+    configs.mail.mailFeatureFlag
+      ? sendMail(transporter, {
```

Comment:
avoid using the same method name twice for different things

Reply:
It looks like it's a recursive function like this

Reply:
Make sense! I thunk you read wrong: the function name is sendEmail (with E), but the function name imported from package is sendMail (without E). However, I'm making the change because it could cause confusion!